lib/manager/gradle-lite/utils.ts

```diff
@@ -0,0 +1,138 @@
+import upath from 'upath';
+import { regEx } from '../../util/regex';
+import { PackageDependency } from '../common';
+import {
+  ManagerData,
+  PackageVariables,
+  Token,
+  TokenType,
+  VariableRegistry,
+} from './common';
+
+const artifactRegex = regEx(
+  '^[a-zA-Z][-_a-zA-Z0-9]*(?:.[a-zA-Z][-_a-zA-Z0-9]*)*$'
+);
+
+const versionLikeRegex = regEx('^(?<version>[-.\\[\\](),a-zA-Z0-9+]+)');
+
+// Extracts version-like and range-like strings
+// from the beginning of input
+export function versionLikeSubstring(input: string): string | null {
+  const match = input ? versionLikeRegex.exec(input) : null;
+  return match ? match.groups.version : null;
+}
+
+export function isDependencyString(input: string): boolean {
+  const split = input?.split(':');
+  if (split?.length !== 3) {
+    return false;
+  }
+  const [groupId, artifactId, versionPart] = split;
+  return (
+    groupId &&
+    artifactId &&
+    versionPart &&
+    artifactRegex.test(groupId) &&
+    artifactRegex.test(artifactId) &&
+    versionPart === versionLikeSubstring(versionPart)
+  );
+}
```

Comment:
This doesn't match the alternative form of `group: "com.example", name: "my.dependency", version: "1.2.3"`

Reply:
> `group: "com.example", name: "my.dependency", version: "1.2.3"`

Can these named parameters can go in any order?

Reply:
yes, i think so. If i remember correctly it's like a hashtable, so order doesn't matter

Reply:
Well, I think it's out of scope for now